src/ValidatorChain.php

```diff
@@ -69,11 +81,9 @@ private function attachValidator($event, $callback, $priority)
             array_unshift($callback, $test);
         }
         if ($context instanceof ValidatorInterface) {
-            $data = $context->getData();
             $name = $context->getName();
-            $this->getStorage()->setMetadata('_VALID', [$name => $data]);
+            $this->getStorage()->setMetadata('_VALID', [$name => serialize($context->current)]);
```

Comment:
Can you check - does `setMetadata` merge or replace data? If it's a replace operation (which its name suggests) then there could only ever be 1 validator in a re-constituted chain.

I don't really understand why the validators are persisted to metadata at all, as they are constructed inside `SessionManager` from a list of configuration items right?